test/widgets/home_test.dart

```diff
@@ -0,0 +1,409 @@
+      await prepare(tester);
+      await tapOpenMenu(tester);
+
+      await tester.tap(channelsMenuIconFinder);
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+      check(find.byType(SingleChildScrollView)).findsNothing();
+    });
+
+    testWidgets('cancel button dismisses the menu', (tester) async {
+      await prepare(tester);
+      await tapOpenMenu(tester);
+
+      await tester.tap(find.text('Cancel'));
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+      check(find.byType(SingleChildScrollView)).findsNothing();
+    });
+
+    testWidgets('menu buttons dismiss the menu', (tester) async {
+      addTearDown(testBinding.reset);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+
+      await tester.pumpWidget(const ZulipApp());
+      await tester.pumpAndSettle();
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      final connection = store.connection as FakeApiConnection;
+
+      await tapOpenMenu(tester);
+
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [eg.streamMessage()]).toJson());
+      await tester.tap(combinedFeedMenuIconFinder);
+      await tester.pump(Duration.zero); // tap the button
+      await tester.pump(const Duration(milliseconds: 250)); // wait for animation
+
+      // When we go back to the home page, the menu sheet should have gone.
+      (await ZulipApp.navigator).pop();
+      await tester.pumpAndSettle();
+      check(find.byType(SingleChildScrollView)).findsNothing();
+    });
+
+    testWidgets('_MyProfileButton', (tester) async {
+      await prepare(tester);
+      await tapOpenMenu(tester);
+
+      await tester.tap(find.text('My profile'));
+      await tester.pumpAndSettle();
+      check(find.byType(ProfilePage)).findsOne();
+      check(find.text(eg.selfUser.fullName)).findsAny();
+    });
+  });
+
+  group('_LoadingPlaceholderPage', () {
+    const loadPerAccountDuration = Duration(seconds: 30);
+    assert(loadPerAccountDuration > kTryAnotherAccountWaitPeriod);
+
+    void checkOnLoadingPage() {
+      check(find.byType(CircularProgressIndicator).hitTestable()).findsOne();
+      check(find.byType(ChooseAccountPage)).findsNothing();
+      check(find.byType(HomePage)).findsNothing();
+    }
+
+    void checkOnChooseAccountPage() {
+      // Ignore the possible loading page in the background.
+      check(find.byType(CircularProgressIndicator).hitTestable()).findsNothing();
+      check(find.byType(ChooseAccountPage)).findsOne();
+      check(find.byType(HomePage)).findsNothing();
+    }
+
+    ModalRoute<dynamic>? getRouteOf(WidgetTester tester, Finder elementFinder) =>
+      ModalRoute.of(tester.element(elementFinder));
```

Comment:
nit: it's just the normal type of finder, no need to be more specific than usual:
```suggestion
    ModalRoute<void>? getRouteOf(WidgetTester tester, Finder finder) =>
      ModalRoute.of(tester.element(finder));
```

(and separately, avoid dynamic)